### DISCUSSION TITLE
Link component right click to open new tab

### DISCUSSION THREAD

**Discussion by anandtripathi5** [vercel/next.js#35529] [Help] [answered]
So basically I want to use the Link left-click functionality as-is. For example, on the left click on the component, it should navigate to a specific page without reloading or refreshing the whole page. But is it possible that on right-click of the component I can use anchor tag property like the open link in new tab?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

SPA routing is a bit of a make-belief situation.

Most SPA techniques require you to use their anchor, to let their internal state keep track of pages, etc.

In the Next.js case, the `next/link` component, needs to wrap an actual anchor tag:

```jsx
<Link href="/">
 <a>anchor</a>
</Link>
```

And it will render an anchor tag with its `href`, with natural browser behavior, including context menus, etc. 

For example on MacOS cmd and tap on track pad opens links on a new tab, and the Link + anchor tag combo does just that.

**Reply by anandtripathi5** [reply]
This will disable link functionality I guess and redirect the page to "/" url. I don't want that to happen on left click only on right-click if possible. Currently I have a workaround to use anchor tag with "#" as href but it is adding # in the url that is annoying.

`<a href="#">anchor</a>`

**Reply by icyJoseph** [reply]
Ah, wait, could you phrase your question different? Maybe step by step? It is a bit hard to read, and understand the last sentences on the original question.

Let me see if I get it, you want to **_disable_** the natural behavior, but keep the ability to open links in new tabs?

Or do you want to keep the natural behavior with left click, but for right clicks you want to do something else? What should happen on **right** click? (Consider people who use devices with pointing capabilities, how should those work)

**Reply by anandtripathi5** [reply]
So when we are using the Link component, we don't want to refresh the whole page just to route to that page without reloading or refreshing. I want the same thing but the downside of it is when you right click on the component it will give you something like this image 
![image](https://user-images.githubusercontent.com/10723482/159527121-12635f13-a576-48b0-b10f-adde6ad30b1e.png)

But I want to open the link in new tab for those component. Like this 
![image](https://user-images.githubusercontent.com/10723482/159527270-f43e789a-a323-4991-9560-75bb519f982c.png)

So basically same functionality of left click but enable open link in new tab on right click

**Reply by icyJoseph** [reply]
Aha! You are misusing the link then.

`next/link` has to wrap an anchor tag. What is your `next/link` component wrapping? How are you using it? If you are wrapping a custom component you need to use `passHref` in `next/link`

## **this is how you use the Link component!** it needs to wrap an `a` tag!

```jsx
<Link href="/some-random-path">
   <a>Hello</a>
</Link>
```

## With a custom component

```jsx
// It has to wrap an `anchor` tag!
const RedLink = styled.a`
  color: red;
`

function NavLink({ href, name }) {
  // Must add passHref to Link
  return (
    <Link href={href} passHref>
      <RedLink>{name}</RedLink>
    </Link>
  )
}
```

Again, this is because of how SPA routing works, read the documentation carefully https://nextjs.org/docs/api-reference/next/link, the `<Link href="some-href"></Link>` by itself will not work! You need to include an anchor tag as children.

---

This is a screenshot of a site I made, the highlight is a `next/link`, you click it and it does the expected behaviour, if you right click on it, it opens the context menu. **Notice** that it has `children: <a/>`

<img width="829" alt="Screenshot 2022-03-22 at 17 54 29" src="https://user-images.githubusercontent.com/21013447/159533658-3cb012eb-e0c9-48de-b488-f4f2af4bb59e.png">

If you could share some code snippet or reproduction to point the error that would be great.

--- 

## Edit

I've tried out now, and I am pretty sure you are doing something like:

```tsx
// ❌ ❌ ❌ ❌
  <Link href="/">
            <span>Home</span>
 </Link>
```

But should do this instead

```tsx
// ✅✅✅
  <Link href="/">
           <a><span>Home</span></a>
 </Link>
```

**Reply by anandtripathi5** [reply]
Its clear now. Thanks for the response

**Reply by icyJoseph** [reply]
Awesome! if possible, please do mark this as answered. And glad to help out more if there's anything else.

**Reply by anandtripathi5** [reply]
Yeah done, can you also add the above reply in answer. As the above reply is having a detailed answer, it will help others too.
